azurerm/internal/services/databricks/databricks_workspace_resource.go

```diff
@@ -114,20 +165,58 @@ func resourceDatabricksWorkspace() *pluginsdk.Resource {
 							AtLeastOneOf: workspaceCustomParametersString(),
 						},
 
+						"public_subnet_network_security_group_association_id": {
+							Type:         pluginsdk.TypeString,
+							Optional:     true,
+							ValidateFunc: azure.ValidateResourceID,
+							AtLeastOneOf: workspaceCustomParametersString(),
+						},
+
 						"private_subnet_name": {
 							Type:         pluginsdk.TypeString,
 							ForceNew:     true,
 							Optional:     true,
 							AtLeastOneOf: workspaceCustomParametersString(),
 						},
 
+						"private_subnet_network_security_group_association_id": {
+							Type:         pluginsdk.TypeString,
+							Optional:     true,
+							ValidateFunc: azure.ValidateResourceID,
+							AtLeastOneOf: workspaceCustomParametersString(),
+						},
+
 						"virtual_network_id": {
 							Type:         pluginsdk.TypeString,
 							ForceNew:     true,
 							Optional:     true,
 							ValidateFunc: azure.ValidateResourceIDOrEmpty,
 							AtLeastOneOf: workspaceCustomParametersString(),
 						},
+
+						"storage_account_name": {
+							Type:         pluginsdk.TypeString,
+							ForceNew:     true,
+							Optional:     true,
+							Computed:     true,
+							AtLeastOneOf: workspaceCustomParametersString(),
+						},
+
+						"storage_account_sku_name": {
```

Comment:
can we get this from the storage account or is it auto created from the name above:

Reply:
No, this is auto created by the Databricks RP, it has a default value of standard_GRS I think. This is offered as an override so you can control sku of the auto generated storage account.